### PR TITLE
Remove phrase 'Abstract' from abstract class names

### PR DESCRIPTION
**Pull request by darthegg**
[Issue #131 ](https://github.com/medipixel/rl_algorithms/issues/131)
Author: Chaehyeuk Lee

1) Remove phrase 'Abstract' from abstract Agent class (algorithms.common.abstract.agent)
2) Remove phrase 'Abstract' from abstract HER class (algorithms.common.abstract.her)
3) Remove phrase 'Abstract' from abstract RewardFn class (algorithms.common.abstract.reward_fn)

Error : Class name is same with Abstract Class name
          ex)   class Agent(Agent)

Fix : Change name of Agent class to include algorithm name
          ex)  class DQNAgent(Agent) 
        Change input name 'HER' to 'her'

Test all algorithms and Environments running

### CODE REVIEW

File: examples/reacher_v2/utils.py [hunk 1]
```diff
@@ -7,11 +7,11 @@
 
 import numpy as np
 
-from algorithms.common.abstract.her import AbstractHER
-from algorithms.common.abstract.reward_fn import AbstractRewardFn
+from algorithms.common.abstract.her import HER as AbstractHER
```
Comment (Curt-Park) [inline examples/reacher_v2/utils.py]: Why HER's class name is declared as AbstractHER, while RewardFn is not AbstractRewardFn?

**Comment by Curt-Park** [comment]
1. Link the issue number.
2. would be good if you add a little description.
3. The PR title will possibly make a misunderstanding. More details needed.
4. Have you checked all algorithms run well?

**Comment by Curt-Park** [comment]
How about "Remove phrase 'Abstract' from abstract class names"?

File: algorithms/fd/dqn_agent.py [hunk 2]
```diff
@@ -18,12 +18,12 @@
...
 from algorithms.common.buffer.replay_buffer import NStepTransitionBuffer
 import algorithms.common.helper_functions as common_utils
-from algorithms.dqn.agent import Agent as DQNAgent
+from algorithms.dqn.agent import DQNAgent
 
 device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
 
 
-class Agent(DQNAgent):
+class DQNfDAgent(DQNAgent):
```
Comment (Curt-Park) [inline algorithms/fd/dqn_agent.py]: DQfD is the correct name.

File: algorithms/per/ddpg_agent.py [hunk 3]
```diff
@@ -13,12 +13,12 @@
...
 from algorithms.common.buffer.priortized_replay_buffer import PrioritizedReplayBuffer
 import algorithms.common.helper_functions as common_utils
-from algorithms.ddpg.agent import Agent as DDPGAgent
+from algorithms.ddpg.agent import DDPGAgent
 
 device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
 
 
-class Agent(DDPGAgent):
+class DDPGPERAgent(DDPGAgent):
```
Comment (Curt-Park) [inline algorithms/per/ddpg_agent.py]: Why not PERDDPG?
Think of the example file (per-ddpg.py) and other similar cases, e.g. bc.

**Comment by darthegg** [comment]
스크린샷, 2019-04-11 14-58-07

File: algorithms/fd/dqn_agent.py (hunk 2 above)
Comment (darthegg) [reply algorithms/fd/dqn_agent.py]: Changed DQNfD to DQfD.

File: algorithms/per/ddpg_agent.py (hunk 3 above)
Comment (darthegg) [reply algorithms/per/ddpg_agent.py]: Changed DDPGPER to PERDDPG

File: README.md [hunk 4]
```diff
@@ -110,8 +110,9 @@ python <run-file> -h
     - Load the saved models and optimizers at the beginning.
 
 ### Class Diagram
-Class diagram drawn on [e447f3e](https://github.com/medipixel/rl_algorithms/commit/e447f3e743f6f85505f2275b646e46f0adcf8f89). This won't be frequently updated.
-![rl_algorithms_cls](https://user-images.githubusercontent.com/14961526/55703648-26022a80-5a15-11e9-8099-9bbfdffcb96d.png)
+Class diagram drawn on [bd76239] (https://github.com/medipixel/rl_algorithms/pull/135/commits/bd76239684d55a92893106a3ceee9cde90294b4d)
```
Comment (Curt-Park) [inline README.md]: The hyperlink to Pull Request would be better rather than commit id; This commit id is not the one merged on master.

**Pull request merged**